### ISSUE TITLE
down sampling op in "_make_block()"

### ISSUE THREAD

**Issue by TanFluent** [QiaoranC/tf_ResNeSt_RegNet_model#6]
hi,  thank you for your amazing jobs on ResNeSt. I am trying to working on my own jobs based on your codes. but i got a few question:
[1]
 in "_make_block()",  when `stride != 1` , we should down sampling the "short-cut" branch.

but, in "_make_block() line221-234", if self.dilation != 1, no down-sampling is apply. Is it a bug?
https://github.com/QiaoranC/tf_ResNeSt_RegNet_model/blob/7cac639387aa75c1cc10262e47b5008fa8123061/models/ResNest.py#L228

[2] "avd" and "avg_down"
as i understand, "avd" control average_downsampling in residual path, while "avg_down" control short-cut path; why not combile as one "avd"?

[3]why dropout layer is discard?
https://github.com/QiaoranC/tf_ResNeSt_RegNet_model/blob/7cac639387aa75c1cc10262e47b5008fa8123061/models/ResNest.py#L422

**Comment by QiaoranC** [comment]
Hi, Thanks for you carefully check.

1. About `self.dilation`, This does looks suspicious. 
   - Truth is `else` wont work, cause always `self.dilation=1`, `self.dilation` only used in _**typically used in Semantic Segmentation**_ , I should delete it in classification technically. 
   - And it directly comes from the official code, you can found [here](https://github.com/zhanghang1989/ResNeSt/blob/e4c37d9b91672dd2301dac10d1e302f781c7fb06/resnest/torch/resnet.py#L238-L243).  But the original code add some more `self.dilation>1` condition for semantic segmentation, i think i skip it.

2.  About the `avd` and `avg_down`, nothing specious here, it simply from the original set. 
    - Under current set, it can be combined,I through about combine when the rewriting, but there are SO many set/terms need to check with original code, one simplified set only makes it harder to check, I decided leave all the terms/set untouched, 

3. About the `dropout`, again, simply follow the original [set](https://github.com/zhanghang1989/ResNeSt/blob/e4c37d9b91672dd2301dac10d1e302f781c7fb06/resnest/torch/resnet.py#L159), welcome to try it.

**Comment by TanFluent** [comment]
@QiaoranC 
thank you for your quik reply.
btw, i try to rewrite your code base on tf.layer instead of tf.keras.layer. Cus, my project is base on tf.layer, and  trying to train a tf.keras.layer based model with "tf.session" almost drive me crazy, lol

**Comment by QiaoranC** [comment]
@TanFluent Just curiosity, and i am only using tf since tf2.x, (former pytorch user). Does  `tf.session` belong to 1.x and no longer support? I heard tf 1.x was pretty nasty, good luck. ResNest is quite complicated.

**Comment by TanFluent** [comment]
> @TanFluent Just curiosity, and i am only using tf since tf2.x, (former pytorch user). Does `tf.session` belong to 1.x and no longer support? I heard tf 1.x was pretty nasty, good luck. ResNest is quite complicated.

My project is tf1.12 and older, where tf.layer is more popular. Since tf2.x, tf.layer is replaced by tf.keras.layer and tf.session is no longer recomand. So, ... ... LOL
